### ISSUE TITLE
ResolveVisualSceneStateSystem overhead

### ISSUE THREAD

**Issue by dalkia** [decentraland/unity-explorer#2694]
While you traverse the world, `SceneDefinitionComponent` keeps getting cached in ECS over and over for unloaded scenes. This can create an overhead both in memory and `ResolveVisualSceneStateSystem` execution, since its iterating through all unloaded scenes.

Start
![image](https://github.com/user-attachments/assets/d35f32cd-64bd-47e3-8311-ec39b838316b)

After walking till a minigame

![image](https://github.com/user-attachments/assets/a6acc1f3-23f7-410e-9903-e6da3453c19e)

And it just keep growing linearly

Its worth checking if we have space to unload `SceneEntityDefinitionComponent` after a certain threshold or distance to unloaded scene.

**Comment by AnsisMalins** [comment]
Ctrl+Shift+F does not find SceneEntityDefinitionComponent. Where is it?

**Comment by dalkia** [comment]
Yes @AnsisMalins, the component is indeed `SceneDefinitionComponent`. Sorry for the confusion

**Comment by AnsisMalins** [comment]
I'm wanting to rewrite the involved systems because I find what I think are other problems with them. For example, they work in 2D, so IsBehind will be wrong for a camera lying on the ground near a parcel and looking up. The top of the parcel could still be visible even if all of its corners on the ground are not. Let me know if I should not do this for any reason.

**Comment by m3taphysics** [comment]
Done.
